internal/services/notification/action/runwebhookdelivery.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2023 Sorint.lab
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package action
+
+import (
+	"context"
+
+	"github.com/sorintlab/errors"
+
+	"agola.io/agola/internal/sqlg/sql"
+	"agola.io/agola/services/notification/types"
+)
+
+type GetRunWebhookDeliveriesRequest struct {
+	ProjectID            string
+	DeliveryStatusFilter []types.DeliveryStatus
+
+	StartSequence uint64
+
+	Limit         int
+	SortDirection types.SortDirection
+}
+
+type GetRunWebhooksDeliveriesResponse struct {
+	RunWebhookDeliveries []*types.RunWebhookDelivery
+
+	HasMore bool
+}
+
+func (h *ActionHandler) GetRunWebhookDeliveries(ctx context.Context, req *GetRunWebhookDeliveriesRequest) (*GetRunWebhooksDeliveriesResponse, error) {
```

Comment:
It's runwebhook deliveries by project so the types and functions names should reflect this.